pfsimulator/parflow_lib/well_package.c

```diff
@@ -510,17 +569,30 @@ void         WellPackage(
             rz = 0;
 
             process = amps_Rank(amps_CommWorld);
-
             new_subgrid = NewSubgrid(ix, iy, iz_lower,
                                      nx, ny, nz,
                                      rx, ry, rz,
                                      process);
             dx = SubgridDX(new_subgrid);
             dy = SubgridDY(new_subgrid);
             dz = SubgridDZ(new_subgrid);
-
-            subgrid_volume = (nx * dx) * (ny * dy) * (nz * dz);
-
+            if (public_xtra->correct_for_var_dz)
+            {
+              if (SubgridLivesOnThisRank(new_subgrid, grid))
+              {
+                subgrid_volume = CalculateSubgridVolume(new_subgrid, problem_data);
+              }
+            }
+            else
+            {
+              subgrid_volume = nx * ny * nz * dx * dy * dz;
+            }
+#ifdef PARFLOW_HAVE_MPI
+            // here I am making the assumption that indices need to be positive so we can use amps_max
```

Comment:
What does the indices have to do with this reduction?   volumes > 0 so find use max?

Reply:
I believe that was my initial concern with this comment but after reviewing the code I don't believe this assumption is required anymore so will remove the comment.